R/Array.R

```diff
@@ -136,6 +136,29 @@ tiledb_array_is_open <- function(arr) {
   libtiledb_array_is_open(arr@ptr)
 }
 
+##' Test if TileDB Array is open for reading
+##'
+##' @param arr A TileDB Array object as for example returned by `tiledb_array()`
+##' @return A boolean indicating whether the TileDB Array object is open, and
+##' the mode is "READ".
+##' @export
+tiledb_array_is_open_for_reading <- function(arr) {
+  stopifnot("The 'arr' argument must be a tiledb_array object" = .isArray(arr))
+  libtiledb_array_is_open(arr@ptr) && tiledb:::libtiledb_array_query_type(arr@ptr) == "READ"
+  return(libtiledb_array_is_open_for_reading(arr@ptr))
+}
+
+##' Test if TileDB Array is open for writing
+##'
+##' @param arr A TileDB Array object as for example returned by `tiledb_array()`
+##' @return A boolean indicating whether the TileDB Array object is open, and
+##' the mode is "WRITE".
+##' @export
+tiledb_array_is_open_for_writing <- function(arr) {
+  stopifnot("The 'arr' argument must be a tiledb_array object" = .isArray(arr))
+  return(libtiledb_array_is_open_for_reading(arr@ptr))
```

Comment:
IIUC this should be `writing`, not `reading`
```suggestion
  return(libtiledb_array_is_open_for_writing(arr@ptr))
```